---
description: Erreur du compilateur CS0103
title: Erreur du compilateur CS0103
ms.date: 07/20/2015
f1_keywords:
- CS0103
helpviewer_keywords:
- CS0103
ms.assetid: fd1f2104-a945-4dba-8137-8ef869826062
ms.openlocfilehash: 689cebb258fcd75287e8e54630387ae7a6e986c5
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 08/30/2020
ms.locfileid: "89134729"
---
# <a name="compiler-error-cs0103"></a>Erreur du compilateur CS0103

Le nom 'identifier' n’existe pas dans le contexte actuel

 Une tentative a été faite pour utiliser un nom qui n’existe ni dans la classe, ni dans l’[espace de noms](../keywords/namespace.md) ni dans la portée. Vérifiez l’orthographe du nom ainsi que vos directives using et références d’assembly afin de vous assurer que le nom que vous essayez d’utiliser est disponible.

 Cette erreur se produit fréquemment si vous déclarez une variable dans une boucle ou un `try` bloc ou, `if` puis essayez d’y accéder à partir d’un bloc de code englobant ou d’un bloc de code séparé, comme indiqué dans l’exemple suivant :

```csharp
using System;

class MyClass1
{
    public static void Main()
    {
        try
        {
            // The following declaration is only available inside the try block.
            var conn = new MyClass1();
        }
        catch (Exception e)
        {  
            // The following expression causes error CS0103, because variable
            // conn only exists in the try block.
            if (conn != null)
                Console.WriteLine("{0}", e);
        }
    }
}
```

 L’exemple suivant résout l’erreur :

```csharp
using System;

class MyClass2
{
    public static void Main()
    {
        // To resolve the error in the example, the first step is to
        // move the declaration of conn out of the try block. The following
        // declaration is available throughout the Main method.
        MyClass2 conn = null;
        try
        {
            // Inside the try block, use the conn variable that you declared
            // previously.
            conn = new MyClass2();
        }
        catch (Exception e)
        {
            // The following expression no longer causes an error, because
            // the declaration of conn is in scope.
            if (conn != null)
                Console.WriteLine("{0}", e);
        }
    }
}
```
